components/sup/src/manager/service/mod.rs

```diff
@@ -213,7 +213,7 @@ impl Service {
             topology: spec.topology,
             update_strategy: spec.update_strategy,
             config_from: spec.config_from,
-            scheduled_health_check: None,
+            scheduled_health_check: Some(Instant::now()),
```

Comment:
With this change, I don't think `scheduled_health_check` ever really needs to be an `Option`, which can simplify some of the rest of this code (it would require a bit of tweaking to `run_health_check_hook`, though).

It also feels like this needs to be harmonized with `schedule_health_check_at_next_tick`, though, since it is the same intent as that method (though not explicitly stated), but different code.

Reply:
I looked at making it not an option and it breaks the other logic. Maybe there's a way to refactor it more, but we do need something to indicate the difference between a health check which is scheduled but is waiting to be run and no health check being scheduled.

Reply:
Alternatively, we make an operational distinction between the scenarios when we want to "schedule something if nothing is already scheduled earlier" and when we want to "schedule the next invocation, because I just finished running this one". The only place where `None` is used is immediately after we execute the health check hook, and we don't leave that function without scheduling something.

I've got a couple ideas of how we might approach this, but I need to play with them a bit. Also, I can already see how some of this is going to have to change when #6107 lands, so I'm happy to try and address this there instead.

Reply:
> The only place where `None` is used is immediately after we execute the health check hook, and we don't leave that function without scheduling something.

Yeah, you're right. I didn't dig into the code enough, and it's not obvious from this level:
```rust
        self.scheduled_health_check = None;

        if check_result == HealthCheck::Ok {
            self.schedule_routine_health_check();
            debug!(
                "Service ({}) health check is: {}",
                self.spec_ident, check_result
            );
        } else {
            debug!(
                "Service ({}) health check is: {}; scheduling special health check",
                self.spec_ident, check_result
            );
            self.schedule_special_health_check();
        }
```
That `self.scheduled_health_check` is `Some` by the end. Perhaps restructuring that a bit would make it clearer. We should add some unit tests while we're at it, though.